pkgs/development/compilers/llvm/10/llvm.nix

```diff
@@ -121,6 +122,19 @@ in stdenv.mkDerivation (rec {
   ] ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
     "-DCMAKE_CROSSCOMPILING=True"
     "-DLLVM_TABLEGEN=${buildPackages.llvm_10}/bin/llvm-tblgen"
+    (
+      let
+        nativeCC = pkgsBuildBuild.stdenv.cc;
```

Comment:
```suggestion
        nativeCC = pkgsBuildBuild.targetPackages.stdenv.cc;
```

Reply:
I don't understand the difference here. In terms of target triples: if we start at `(x, y, z)`, then `pkgsBuildBuild` takes us to `(*, x, x)`, and `targetPackages` (aka `pkgsTargetTarget`) takes us to `(*, x, x)` so it seems like a no-op. Is there another difference?

Reply:
`stdenv` takes tools from the previous stage, so:

1. `pkgsBuildBuild`: `(?1, x, x)`
2. `pkgsBuildBuild.stdenv.cc`: `(?0, ?1, x)`
 
while:

1. `pkgsBuildBuild`: `(?1, x, x)`
2. `pkgsBuildBuild.targetPackages`: `(x, x, ?2)`
3. `pkgsBuildBuild.targetPackages.stdenv.cc`: `(?1, x, x)`